src/porepy/models/energy_balance.py

```diff
@@ -265,22 +265,29 @@ def interface_energy_flux(self, interfaces: list[pp.MortarGrid]) -> pp.ad.Operat
         flux.set_name("interface_energy_flux")
         return flux
 
-    def mobility_rho_h(self, domains: pp.SubdomainsOrBoundaries) -> pp.ad.Operator:
-        """ "Non-linear weight in the advective enthalpy flux.
+    def advection_weight_energy_balance(
```

Comment:
The version in the CF module has additional treatment for the fractional flow formulation, where the advected enthalpy is given as `h * f_phase * m`, assuming `m` has the total mobility in its second-order tensor.
Otherwise it performs a super-call.

The `TotalMassBalanceEquations` in contrast are not affected by that, because there is no fractional flow of total mass.

The version in the CF module can therefore be used with both, `TotalMassBalanceEquations` and `DiffusiveTotalMassBalanceEquations`.